### PR TITLE
refactor vol_perc to not depend on alsa libraries

### PR DESCRIPTION
**Pull request by parazyd**

### CODE REVIEW

**Comment by drkhsh** [comment]
Very cool! Thanks, I will play with it and add some notes and merge in a few days

**Comment by R41z** [comment]
that's nice, if just you can get a function that works for BSDs, that would be perfect!

**Comment by parazyd** [comment]
@R41z this should work on BSD

File: config.def.h [hunk 1]
```diff
@@ -34,7 +34,7 @@
 - uid (uid of current user) [argument: NULL]
 - uptime (uptime) [argument: NULL]
 - username (username of current user) [argument: NULL]
-- vol_perc (alsa volume and mute status in percent) [argument: soundcard]
+- vol_perc (oss/alsa volume and mute status in percent) [argument: /dev/mixer]
```
Comment (drkhsh) [inline config.def.h]: Maybe "mixer file, /dev/mixer" instead of "/dev/mixer" as comment? Would simplify it for newbies maybe.

Comment (drkhsh) [inline config.def.h]: Maybe improve the argument-comment here?

**Comment by drkhsh** [comment]
@parazyd We should also address the "modprobe snd-pcm-oss" thing in the README.md.

File: slstatus.c [hunk 2]
```diff
@@ -617,41 +617,26 @@ uid(void)
...
+	ioctl(afd, MIXER_READ(SOUND_MIXER_DEVMASK), &devmask);
+	for (i = 0; i < (sizeof(vnames) / sizeof((vnames[0]))); i++)
+		if (devmask & (1 << i))
+			if (!strcmp("vol", vnames[i]))
+				ioctl(afd, MIXER_READ(i), &v);
 
-	if (!mute)
+	close(afd);
+	if (v == 0)
 		return smprintf("mute");
```
Comment (drkhsh) [inline slstatus.c]: could you use braces for single line statements?

File: slstatus.c [hunk 3]
```diff
@@ -617,41 +617,26 @@ uid(void)
...
-	snd_mixer_selem_get_playback_volume(elem, 0, &vol);
-	snd_mixer_selem_get_playback_switch(elem, 0, &mute);
-
-	snd_mixer_selem_id_free(s_elem);
-	snd_mixer_close(handle);
+	ioctl(afd, MIXER_READ(SOUND_MIXER_DEVMASK), &devmask);
+	for (i = 0; i < (sizeof(vnames) / sizeof((vnames[0]))); i++)
+		if (devmask & (1 << i))
+			if (!strcmp("vol", vnames[i]))
+				ioctl(afd, MIXER_READ(i), &v);
```
Comment (drkhsh) [inline slstatus.c]: could you use braces for single line statements?

**Comment by drkhsh** [comment]
@parazyd Thanks for your great work!

**Pull request merged**